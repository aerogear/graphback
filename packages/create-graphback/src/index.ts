#!/usr/bin/env node

import * as yargs from 'yargs';

import { init } from './init/components';

type Params = {
  name?: string;
  templateName?: string;
  templateUrl?: string;
  filter?: string;
};

const command = '$0 <name>';

export const desc = 'Create Graphback project from available templates';

// tslint:disable-next-line: typedef
export const builder = (args: yargs.Argv) => {
  args.positional('name', {
    describe: 'Project name',
    type: 'string'
  });
  args.option('templateName', {
    describe: 'Name of the predefined template',
    type: 'string'
  });
  args.option('templateUrl', {
    describe:
      'GitHub URL of the template. For example (https://github.com/wtrocki/graphql-serve-example#master)',
    type: 'string'
  });
  args.option('filter', {
    describe:
      'A filter option used to limit the number of displayed templates that only matches it. For example "postgres" will only display templates that have a Postgres backed',
    type: 'string'
  });
};

export async function handler({
  name,
  templateName,
  templateUrl,
  filter
}: Params) {
  await init(name, templateName, templateUrl, filter);
}

if (require.main === module) {
  // eslint-disable-next-line no-unused-expressions
  yargs
    .command(command, desc, builder, handler)
    .example(
      '$0 my-awesome-project',
      'creates the "my-awesome-project" in the current working directory'
    )
    .demandCommand(1)
    .strict()
    .recommendCommands()
    .help()
    .alias('h', 'help')
    .version()
    .alias('v', 'version').argv;
}
