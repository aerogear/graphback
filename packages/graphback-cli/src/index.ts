#!/usr/bin/env node

import * as yargs from 'yargs';

if (require.main === module) {
  // eslint-disable-next-line no-unused-expressions
  yargs
    .commandDir('commands')
    .demandCommand(1)
    .strict()
    .recommendCommands()
    .help()
    .alias('h', 'help')
    .version()
    .alias('v', 'version')
    .argv;
}
