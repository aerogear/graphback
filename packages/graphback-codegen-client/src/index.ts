export * from './ClientDocuments'
export * from './ClientCRUDPlugin'