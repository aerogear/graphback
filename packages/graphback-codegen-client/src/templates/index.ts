export { createClientDocumentsTS } from "./tsTemplates"
export * from "./gqlTemplates"