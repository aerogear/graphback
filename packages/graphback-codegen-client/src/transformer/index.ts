import { InputModelTypeContext } from '@graphback/codegen-input';
import {  createClientDocuments } from './targetContext';

/**
 * Generate ClientDocuments from the datamodel.
 * Those definitions will be the actual queries, 
 * mutations and subscriptions that can be then saved to some predefined locations.
 */
export class ClientGenerator {
  private inputContext: InputModelTypeContext[]

  constructor(inputContext: InputModelTypeContext[]) {
    this.inputContext = inputContext
  }

  public generate() {
    return createClientDocuments(this.inputContext)
  }
}
