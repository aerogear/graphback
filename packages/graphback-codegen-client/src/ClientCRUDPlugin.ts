import { GraphbackCoreMetadata, GraphbackPlugin } from '@graphback/core'
import { writeDocumentsToFilesystem } from './helpers/writeDocuments';
import { createClientDocumentsGQL, createClientDocumentsTS } from './templates'
import { ClientTemplates } from './templates/ClientTemplates'

/**
 * Configuration for client generator
 */
export interface ClientGeneratorPluginConfig {
  /**
   * Output language that will be supported
   * Our plugin supports multiple languages for simplicity
   *
   * - ts - typescript file output (backwards compatibility)
   * - graphql - .graphql file
   */
  format: 'ts' | 'graphql'

  /**
   * Generate only fragments and skip query, mutation and subscription elements
   * This can be particulairly usefull when many custom complex queries are needed
   * but they still base on autogenerated fragments
   */
  fragmentOnly?: boolean

  /**
   * RelativePath for the output file created by generator
   */
  outputFile: string
}

export const CLIENT_CRUD_PLUGIN = "ClientCRUDPlugin";

/**
 * Graphback CRUD operations plugin
 *
 * Plugins generates client side documents containing CRUD operations:
 * Queries, Mutations and Subscriptions that reference coresponding schema and resolves.
 * Plugin operates on all types annotated with model
 *
 * Used graphql metadata:
 *
 * - model: marks type to be processed by CRUD generator
 * - crud: controls what types of operations can be generated.
 * For example crud.update: false will disable updates for type
 */
export class ClientCRUDPlugin extends GraphbackPlugin {
  private pluginConfig: ClientGeneratorPluginConfig;

  public constructor(pluginConfig?: ClientGeneratorPluginConfig) {
    super()
    this.pluginConfig = Object.assign({ format: 'graphql' }, pluginConfig);
    if (!pluginConfig.outputFile) {
      throw new Error("client plugin requires outputFile parameter")
    }
  }

  public createResources(metadata: GraphbackCoreMetadata): void {
    const documents = this.getDocuments(metadata)
    const outputFormat = this.pluginConfig.format;

    writeDocumentsToFilesystem(this.pluginConfig.outputFile, documents, outputFormat);
  }

  public getPluginName(): string {
    return CLIENT_CRUD_PLUGIN;
  }

  public getDocuments(metadata: GraphbackCoreMetadata) {
    const models = metadata.getModelDefinitions();
    if (models.length === 0) {
      this.logWarning("Provided schema has no models. No client side queries will be generated")
    }

    let documents: ClientTemplates;

    if (this.pluginConfig.format === 'ts') {
      documents = createClientDocumentsTS(models);
    }
    else if (this.pluginConfig.format === 'graphql') {
      documents = createClientDocumentsGQL(models);
    } else {
      throw new Error("Invalid output format for client plugin");
    }

    if (this.pluginConfig.fragmentOnly) {
      documents = {
        fragments: documents.fragments,
        queries: [],
        mutations:[],
        subscriptions: [],
      };
    }

    return documents;
  }
}
