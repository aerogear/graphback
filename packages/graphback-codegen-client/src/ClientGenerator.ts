import { InputModelTypeContext } from '@graphback/codegen-input';
import { ClientGeneratorConfig } from './createClient';
import { createClientDocumentsGQL } from './templates/gqlTemplates';
import { createClientDocumentsTS } from './templates/tsTemplates';


/**
 * Generate ClientDocuments from the datamodel.
 * Those definitions will be the actual queries, 
 * mutations and subscriptions that can be then saved to some predefined locations.
 */
export class ClientGenerator {
  private inputContext: InputModelTypeContext[]
  private config: ClientGeneratorConfig;

  constructor(inputContext: InputModelTypeContext[], config: ClientGeneratorConfig) {
    this.inputContext = inputContext
    this.config = config;
  }

  public generate() {
    if(this.config){
      if(this.config.language === 'ts'){
        return createClientDocumentsTS(this.inputContext)
      }
      if(this.config.language === 'gql'){
        return createClientDocumentsGQL(this.inputContext)
      }
    }
    
    return createClientDocumentsTS(this.inputContext)
  }
}
