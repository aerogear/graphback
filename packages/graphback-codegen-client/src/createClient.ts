import { InputModelTypeContext } from '@graphback/core';
import { ClientDocuments } from './api/ClientDocuments';
import { ClientGenerator } from './ClientGenerator';


/**
 * Configuration for client generator
 */
export interface ClientGeneratorConfig {
    /**
     * Output language that will be supported
     * Our plugin supports multiple languages for simplicity
     *
     * - ts - typescript file output (backwards compatibility)
     * - gql - .graphql file
     */
    output: 'ts' | 'gql'
}


/**
 * Create client side based on inputContext
 *
 * @param inputContext
 */
export const createClient = async (inputContext: InputModelTypeContext[], config?: ClientGeneratorConfig): Promise<ClientDocuments> => {
    const clientGenerator = new ClientGenerator(inputContext, config);

    return clientGenerator.generate();
}
