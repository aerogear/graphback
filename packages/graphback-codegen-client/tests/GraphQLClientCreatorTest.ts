import { graphQLInputContext } from '@graphback/core';
// tslint:disable-next-line: match-default-export-name no-implicit-dependencies
import ava, { ExecutionContext } from 'ava';
import { readFileSync } from 'fs';
import { parse } from 'graphql';
import { createClient } from '../src';

const schemaText = readFileSync(`${__dirname}/mock.graphql`, 'utf8')

ava('Test snapshot config ts', async (t: ExecutionContext) => {
  const defautConfig = {
    "create": true,
    "update": true,
    "findAll": true,
    "find": true,
    "delete": true,
    "subCreate": true,
    "subUpdate": true,
    "subDelete": true,
    "disableGen": true
  }
  const inputContext = graphQLInputContext.createModelContext(schemaText, defautConfig)
  const client = await createClient(inputContext)
  t.snapshot(client);
});


ava('Test snapshot empty config', async (t: ExecutionContext) => {
  const defautConfig = {
  }
  const inputContext = graphQLInputContext.createModelContext(schemaText, defautConfig)
  const client = await createClient(inputContext)
  t.snapshot(client);
});

ava('Test snapshot config gql', async (t: ExecutionContext) => {
  const defautConfig = {
    "create": true,
    "update": true,
    "findAll": true,
    "find": true,
    "delete": true,
    "subCreate": true,
    "subUpdate": true,
    "subDelete": true,
    "disableGen": true
  }
  const inputContext = graphQLInputContext.createModelContext(schemaText, defautConfig)
  const client = await createClient(inputContext, { output: "gql" })
  t.snapshot(client);
});

ava('Test snapshot config gqlwithfragmentz', async (t: ExecutionContext) => {
  const defautConfig = {
    "create": true,
    "update": true,
    "findAll": true,
    "find": true,
    "delete": true,
    "subCreate": true,
    "subUpdate": true,
    "subDelete": true,
    "disableGen": true
  }
  const inputContext = graphQLInputContext.createModelContext(schemaText, defautConfig)
  const client = await createClient(inputContext, { output: "gqlwithfragment" })
  t.snapshot(client);
});

ava('Test parse', async (t: ExecutionContext) => {
  const defautConfig = {
    "create": true,
    "update": true,
    "findAll": true,
    "find": true,
    "delete": true,
    "subCreate": true,
    "subUpdate": true,
    "subDelete": true,
    "disableGen": true
  }
  const inputContext = graphQLInputContext.createModelContext(schemaText, defautConfig)
  const client = await createClient(inputContext, { output: "gql" })
  for(const documentObj of client.queries){
    const doc = parse(documentObj.implementation);
    t.true(doc.kind === 'Document');
    // TODO Requries full schema generator
    // validate(schema, doc));
  }
  for(const documentObj of client.mutations){
    const doc = parse(documentObj.implementation);
    t.true(doc.kind === 'Document');
    // TODO Requries full schema generator
    // validate(schema, doc));
  }
  for(const documentObj of client.subscriptions){
    const doc = parse(documentObj.implementation);
    t.true(doc.kind === 'Document');
    // TODO Requries full schema generator
    // validate(schema, doc));
  }
});


 