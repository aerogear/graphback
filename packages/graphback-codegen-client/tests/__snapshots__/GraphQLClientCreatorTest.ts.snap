// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test plugin engine gql 1`] = `
Object {
  "fragments": Array [
    Object {
      "implementation": "fragment NoteFields on Note {
   id
   title
   description

} ",
      "name": "Note",
    },
    Object {
      "implementation": "fragment NoteExpandedFields on Note {
   id
   title
   description
   comments {
      id
      title
      description
   }
} ",
      "name": "NoteExpanded",
    },
    Object {
      "implementation": "fragment CommentFields on Comment {
   id
   title
   description

} ",
      "name": "Comment",
    },
    Object {
      "implementation": "fragment CommentExpandedFields on Comment {
   id
   title
   description
   noteComment {
      id
      title
      description
   }
} ",
      "name": "CommentExpanded",
    },
  ],
  "mutations": Array [
    Object {
      "implementation": "mutation createNote($input: NoteInput!) {
  createNote(input: $input) {
      ...NoteFields
  }
}
",
      "name": "createNote",
    },
    Object {
      "implementation": "mutation updateNote($input: NoteInput!) {
  updateNote(input: $input) {
      ...NoteFields
  }
}
",
      "name": "updateNote",
    },
    Object {
      "implementation": "mutation deleteNote($input: NoteInput!) {
  deleteNote(input: $input) {
      ...NoteFields
  }
}
",
      "name": "deleteNote",
    },
    Object {
      "implementation": "mutation createComment($input: CommentInput!) {
  createComment(input: $input) {
      ...CommentFields
  }
}
",
      "name": "createComment",
    },
    Object {
      "implementation": "mutation updateComment($input: CommentInput!) {
  updateComment(input: $input) {
      ...CommentFields
  }
}
",
      "name": "updateComment",
    },
    Object {
      "implementation": "mutation deleteComment($input: CommentInput!) {
  deleteComment(input: $input) {
      ...CommentFields
  }
}
",
      "name": "deleteComment",
    },
  ],
  "queries": Array [
    Object {
      "implementation": "query findNotes($fields: NoteInput!, $limit: Int, $offset: Int) {
    findNotes(fields: $fields, limit: $limit, offset: $offset) {
      ...NoteExpandedFields
    }
  }",
      "name": "findNotes",
    },
    Object {
      "implementation": "query findAllNotes($limit: Int, $offset: Int) {
    findAllNotes(limit: $limit, offset: $offset) {
      ...NoteExpandedFields
    }
  }",
      "name": "findAllNotes",
    },
    Object {
      "implementation": "query findComments($fields: CommentInput!, $limit: Int, $offset: Int) {
    findComments(fields: $fields, limit: $limit, offset: $offset) {
      ...CommentExpandedFields
    }
  }",
      "name": "findComments",
    },
    Object {
      "implementation": "query findAllComments($limit: Int, $offset: Int) {
    findAllComments(limit: $limit, offset: $offset) {
      ...CommentExpandedFields
    }
  }",
      "name": "findAllComments",
    },
  ],
  "subscriptions": Array [
    Object {
      "implementation": "subscription newNote {
  newNote {
      ...NoteFields
  }
} ",
      "name": "newNote",
    },
    Object {
      "implementation": "subscription updatedNote {
  updatedNote {
      ...NoteFields
  }
} ",
      "name": "updatedNote",
    },
    Object {
      "implementation": "subscription deletedNote {
  deletedNote {
      ...NoteFields
  }
} ",
      "name": "deletedNote",
    },
    Object {
      "implementation": "subscription newComment {
  newComment {
      ...CommentFields
  }
} ",
      "name": "newComment",
    },
    Object {
      "implementation": "subscription updatedComment {
  updatedComment {
      ...CommentFields
  }
} ",
      "name": "updatedComment",
    },
    Object {
      "implementation": "subscription deletedComment {
  deletedComment {
      ...CommentFields
  }
} ",
      "name": "deletedComment",
    },
  ],
}
`;

exports[`Test plugin engine gqlfragments 1`] = `
Object {
  "fragments": Array [],
  "mutations": Array [
    Object {
      "implementation": "

mutation createNote($input: NoteInput!) {
  createNote(input: $input) {
      ...NoteFields
  }
}


fragment NoteFields on Note {
   id
   title
   description

} 

",
      "name": "createNote",
    },
    Object {
      "implementation": "

mutation updateNote($input: NoteInput!) {
  updateNote(input: $input) {
      ...NoteFields
  }
}


fragment NoteFields on Note {
   id
   title
   description

} 

",
      "name": "updateNote",
    },
    Object {
      "implementation": "

mutation deleteNote($input: NoteInput!) {
  deleteNote(input: $input) {
      ...NoteFields
  }
}


fragment NoteFields on Note {
   id
   title
   description

} 

",
      "name": "deleteNote",
    },
    Object {
      "implementation": "

mutation createComment($input: CommentInput!) {
  createComment(input: $input) {
      ...CommentFields
  }
}


fragment CommentFields on Comment {
   id
   title
   description

} 

",
      "name": "createComment",
    },
    Object {
      "implementation": "

mutation updateComment($input: CommentInput!) {
  updateComment(input: $input) {
      ...CommentFields
  }
}


fragment CommentFields on Comment {
   id
   title
   description

} 

",
      "name": "updateComment",
    },
    Object {
      "implementation": "

mutation deleteComment($input: CommentInput!) {
  deleteComment(input: $input) {
      ...CommentFields
  }
}


fragment CommentFields on Comment {
   id
   title
   description

} 

",
      "name": "deleteComment",
    },
  ],
  "queries": Array [
    Object {
      "implementation": "

query findNotes($fields: NoteInput!, $limit: Int, $offset: Int) {
    findNotes(fields: $fields, limit: $limit, offset: $offset) {
      ...NoteExpandedFields
    }
  }

fragment NoteExpandedFields on Note {
   id
   title
   description
   comments {
      id
      title
      description
   }
} 

",
      "name": "findNotes",
    },
    Object {
      "implementation": "

query getNote($filter: NoteFilterUniqueInput) {
    getNote(filter: $filter) {
      ...NoteExpandedFields
    }
  }

fragment NoteExpandedFields on Note {
   id
   title
   description
   comments {
      id
      title
      description
   }
} 

",
      "name": "getNote",
    },
    Object {
      "implementation": "

query findComments($fields: CommentInput!, $limit: Int, $offset: Int) {
    findComments(fields: $fields, limit: $limit, offset: $offset) {
      ...CommentExpandedFields
    }
  }

fragment CommentExpandedFields on Comment {
   id
   title
   description
   noteComment {
      id
      title
      description
   }
} 

",
      "name": "findComments",
    },
    Object {
      "implementation": "

query getComment($filter: CommentFilterUniqueInput) {
    getComment(filter: $filter) {
      ...CommentExpandedFields
    }
  }

fragment CommentExpandedFields on Comment {
   id
   title
   description
   noteComment {
      id
      title
      description
   }
} 

",
      "name": "getComment",
    },
  ],
  "subscriptions": Array [
    Object {
      "implementation": "

subscription newNote {
  newNote {
      ...NoteFields
  }
} 

fragment NoteFields on Note {
   id
   title
   description

} 

",
      "name": "newNote",
    },
    Object {
      "implementation": "

subscription updatedNote {
  updatedNote {
      ...NoteFields
  }
} 

fragment NoteFields on Note {
   id
   title
   description

} 

",
      "name": "updatedNote",
    },
    Object {
      "implementation": "

subscription deletedNote {
  deletedNote {
      ...NoteFields
  }
} 

fragment NoteFields on Note {
   id
   title
   description

} 

",
      "name": "deletedNote",
    },
    Object {
      "implementation": "

subscription newComment {
  newComment {
      ...CommentFields
  }
} 

fragment CommentFields on Comment {
   id
   title
   description

} 

",
      "name": "newComment",
    },
    Object {
      "implementation": "

subscription updatedComment {
  updatedComment {
      ...CommentFields
  }
} 

fragment CommentFields on Comment {
   id
   title
   description

} 

",
      "name": "updatedComment",
    },
    Object {
      "implementation": "

subscription deletedComment {
  deletedComment {
      ...CommentFields
  }
} 

fragment CommentFields on Comment {
   id
   title
   description

} 

",
      "name": "deletedComment",
    },
  ],
}
`;

exports[`Test plugin engine ts 1`] = `
Object {
  "fragments": Array [
    Object {
      "implementation": "import gql from \\"graphql-tag\\"

export const NoteFragment = gql\`
  fragment NoteFields on Note {
   id
   title
   description

} 
\`
",
      "name": "Note",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"

export const NoteExpandedFragment = gql\`
  fragment NoteExpandedFields on Note {
   id
   title
   description
   comments {
      id
      title
      description
   }
} 
\`
",
      "name": "NoteExpanded",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"

export const CommentFragment = gql\`
  fragment CommentFields on Comment {
   id
   title
   description

} 
\`
",
      "name": "Comment",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"

export const CommentExpandedFragment = gql\`
  fragment CommentExpandedFields on Comment {
   id
   title
   description
   noteComment {
      id
      title
      description
   }
} 
\`
",
      "name": "CommentExpanded",
    },
  ],
  "mutations": Array [
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteFragment } from \\"../fragments/Note\\"

export const createNote = gql\`
  mutation createNote($input: NoteInput!) {
  createNote(input: $input) {
      ...NoteFields
  }
}


  \${NoteFragment}
\`
",
      "name": "createNote",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteFragment } from \\"../fragments/Note\\"

export const updateNote = gql\`
  mutation updateNote($input: NoteInput!) {
  updateNote(input: $input) {
      ...NoteFields
  }
}


  \${NoteFragment}
\`
",
      "name": "updateNote",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteFragment } from \\"../fragments/Note\\"

export const deleteNote = gql\`
  mutation deleteNote($input: NoteInput!) {
  deleteNote(input: $input) {
      ...NoteFields
  }
}


  \${NoteFragment}
\`
",
      "name": "deleteNote",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentFragment } from \\"../fragments/Comment\\"

export const createComment = gql\`
  mutation createComment($input: CommentInput!) {
  createComment(input: $input) {
      ...CommentFields
  }
}


  \${CommentFragment}
\`
",
      "name": "createComment",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentFragment } from \\"../fragments/Comment\\"

export const updateComment = gql\`
  mutation updateComment($input: CommentInput!) {
  updateComment(input: $input) {
      ...CommentFields
  }
}


  \${CommentFragment}
\`
",
      "name": "updateComment",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentFragment } from \\"../fragments/Comment\\"

export const deleteComment = gql\`
  mutation deleteComment($input: CommentInput!) {
  deleteComment(input: $input) {
      ...CommentFields
  }
}


  \${CommentFragment}
\`
",
      "name": "deleteComment",
    },
  ],
  "queries": Array [
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteExpandedFragment } from \\"../fragments/NoteExpanded\\"

export const findNotes = gql\`
  query findNotes($fields: NoteInput!, $limit: Int, $offset: Int) {
    findNotes(fields: $fields, limit: $limit, offset: $offset) {
      ...NoteExpandedFields
    }
  }

  \${NoteExpandedFragment}
\`
",
      "name": "findNotes",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteExpandedFragment } from \\"../fragments/NoteExpanded\\"

export const getNote = gql\`
  query getNote($filter: NoteFilterUniqueInput) {
    getNote(filter: $filter) {
      ...NoteExpandedFields
    }
  }

  \${NoteExpandedFragment}
\`
",
      "name": "getNote",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentExpandedFragment } from \\"../fragments/CommentExpanded\\"

export const findComments = gql\`
  query findComments($fields: CommentInput!, $limit: Int, $offset: Int) {
    findComments(fields: $fields, limit: $limit, offset: $offset) {
      ...CommentExpandedFields
    }
  }

  \${CommentExpandedFragment}
\`
",
      "name": "findComments",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentExpandedFragment } from \\"../fragments/CommentExpanded\\"

export const getComment = gql\`
  query getComment($filter: CommentFilterUniqueInput) {
    getComment(filter: $filter) {
      ...CommentExpandedFields
    }
  }

  \${CommentExpandedFragment}
\`
",
      "name": "getComment",
    },
  ],
  "subscriptions": Array [
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteFragment } from \\"../fragments/Note\\"

export const newNote = gql\`
  subscription newNote {
  newNote {
      ...NoteFields
  }
} 

  \${NoteFragment}
\`
",
      "name": "newNote",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteFragment } from \\"../fragments/Note\\"

export const updatedNote = gql\`
  subscription updatedNote {
  updatedNote {
      ...NoteFields
  }
} 

  \${NoteFragment}
\`
",
      "name": "updatedNote",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { NoteFragment } from \\"../fragments/Note\\"

export const deletedNote = gql\`
  subscription deletedNote {
  deletedNote {
      ...NoteFields
  }
} 

  \${NoteFragment}
\`
",
      "name": "deletedNote",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentFragment } from \\"../fragments/Comment\\"

export const newComment = gql\`
  subscription newComment {
  newComment {
      ...CommentFields
  }
} 

  \${CommentFragment}
\`
",
      "name": "newComment",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentFragment } from \\"../fragments/Comment\\"

export const updatedComment = gql\`
  subscription updatedComment {
  updatedComment {
      ...CommentFields
  }
} 

  \${CommentFragment}
\`
",
      "name": "updatedComment",
    },
    Object {
      "implementation": "import gql from \\"graphql-tag\\"
import { CommentFragment } from \\"../fragments/Comment\\"

export const deletedComment = gql\`
  subscription deletedComment {
  deletedComment {
      ...CommentFields
  }
} 

  \${CommentFragment}
\`
",
      "name": "deletedComment",
    },
  ],
}
`;
